### ISSUE TITLE
Map column names recipe doesn't seem to work

### ISSUE THREAD

**Issue by davewasmer** [Vincit/objection.js#290]
The [recipe for mapping column names](http://vincit.github.io/objection.js/#map-column-names-to-different-property-names) seems to indicate that the `$parseDatabaseJson` method will get a POJO representing a single row (i.e. the examples use `_.mapKeys` on the incoming object directly).

But setting this up myself, it seems like my `$parseDatabaseJson` method is getting a `Result` object as the argument, which looks like it represents the entire database response, rather than a single row:

![image](https://cloud.githubusercontent.com/assets/707213/22526218/9fba6132-e887-11e6-8d1b-fab24e8e4155.png)

I checked and can't seem to find a Result class in Objection or Knex, so not sure where that's coming from. Is there any documentation for this Result class?

**Comment by fl0w** [comment]
From the picture, this looks to be the result set that `knex.js` returns. Not close to a computer so can't check this at the moment though.

**Comment by devinivy** [comment]
Those are the results provided by node-postgres!  See https://github.com/brianc/node-postgres/blob/83a946f61cb9e74c7f499e44d03a268c399bd623/lib/result.js#L14

**Comment by davewasmer** [comment]
So is this the intended behavior for Objection (to return whatever the driver returns, rather than trying to normalize across drivers)? Or is this a bug?

**Comment by koskimas** [comment]
No this is not intended. Knex takes care of normalizing the data, and whatever knex returns is passed to `$parseDatabaseJson` method. In some cases (raw queries) knex fails to parse the db response. @davewasmer Could you post some code of the query you are performing?

**Comment by davewasmer** [comment]
Looks like it was a coincidence of a few different issues at the same time. I was trying to do a simple `MyModel.query().insert(data)`, but:

* My `$formatDatabaseJson()` method that fired on that query was accidentally returning `undefined` (:man_facepalming: I know, I know, it was late lol)
* This caused knex to treat the insert as a noop and silently do nothing (see [this comment](https://github.com/tgriesser/knex/issues/589#issuecomment-94904428))
* I'm guessing that because the query was a noop, it returns the raw driver result

So definitely user error (a.k.a my fault) on returning `undefined`, but the behavior of Knex/Objection was surprising here and made it very difficult to track down what was going on. I ended up having to set breakpoints deep in the Knex codebase to notice that when it tried to build the insert query, it just set the query to an empty string. From there, I tracked down the above-linked issue, and realized it was noop-ing the query because the insert value was undefined.

Not sure if there's anything to be done on the Objection side here, but just wanted to offer the feedback from a user perspective. Thanks for all your work on this lib btw!
